onnx/shape_inference/implementation.cc

```diff
@@ -563,6 +563,10 @@ class ShapeInferenceImplBase {
             ir_version,
             model_local_functions_map} {}
 
+  std::unordered_map<std::string, TensorShapeProto> getGeneratedShapeData() {
+    return generated_shape_data_by_name;
```

Comment:
Seems okay for now. But this may end up creating unnecessary copies of a large data-structure. It may be useful to consider a design that uses a `std::move` or some other approach.

Reply:
Good point. Actually I was considering to modify `generated_shape_data_by_name` in place without returning it by `InferShapesAndDataPropagation` since another input parameter `ModelProto` is also called by reference and can be modified in place. So here are two options:
1. Make `generated_shape_data_by_name` call by reference as `InferShapesAndDataPropagation`'s input.
2. Return `generated_shape_data_by_name` additionally by `InferShapesAndDataPropagation`. (What this PR currently does and it needs some improvement for moving it as you mentioned)

@gramalingam do you have a preference about it? Thanks.

Reply:
There are two interfaces where copying can happen: when the top-level `Infer...` API is called, and second when the Base class constructor is called. So, both will need to be addressed. 

I think that if option 1 works, that should be good.

Reply:
Specifically, the question is: can we also change the base-class constructor to take a non-const reference, and avoid copying there?

Reply:
Thanks for your opinion. I made base-class constructor take a non-const reference and `InferShapesAndDataPropagation` use `generated_shape_data_by_name` by reference, which can address some potential copy issues. Please review it again.